executor/write_test.go

```diff
@@ -1444,6 +1444,14 @@ func (s *testSuite8) TestUpdate(c *C) {
 	r.Check(testkit.Rows("Warning 1062 Duplicate entry '1' for key 'I_uniq'"))
 	tk.MustQuery("select * from t").Check(testkit.Rows("1", "2"))
 
+	// test issue21965
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("create table t (a int) partition by list (a) (partition p0 values in (0,1));")
+	tk.MustExec("insert ignore into t values (1);")
+	_, err = tk.Exec("update ignore t set a=2 where a=1;")
```

Comment:
Could you use `tk.MustExec` here?

Reply:
Yes, I have changed it to `tk.MustExec`.